cartool/main.m

```diff
@@ -134,32 +131,65 @@ void CGImageWriteToFile(CGImageRef image, NSString *path)
 {
     NSMutableArray *images = [[NSMutableArray alloc] initWithCapacity:5];
 
-    for (NSNumber *scaleFactor in @[@1, @2, @3])
-    {
-        CUINamedImage *image = [catalog imageWithName:key scaleFactor:scaleFactor.doubleValue];
+    
+    struct CGPDFDocument *pdfDocument = [catalog pdfDocumentWithName:key];
+    
+    if (pdfDocument) {
+        [images addObject:(__bridge id _Nonnull)(pdfDocument)];
+    } else {
+        for (NSNumber *scaleFactor in @[@1, @2, @3])
+        {
+            CUINamedImage *image = [catalog imageWithName:key scaleFactor:scaleFactor.doubleValue];
 
-        if (image && image.scale == scaleFactor.floatValue) [images addObject:image];
+            if (image && image.scale == scaleFactor.floatValue) [images addObject:image];
+        }
     }
-
     return images;
 }
 
+void writePDFtoFile(struct CGPDFDocument *pdfDocument, NSString *path, NSString *key) {
```

Comment:
I'm not super familiar with `CGPDFDocument`, but I cobbled this together from a few answers on Stack Overflow and pastern and it hasn't failed me yet